### DISCUSSION TITLE
Accept functions in to support dynamic variants

### DISCUSSION THREAD

**Discussion by xndyz** [joe-bell/cva#56] [Ideas]
It's really nice to be able to pass in a dynamic variant like "color=gray-500" and render "text-gray-500". This is esp useful with Tailwind which has a large utility map built-in, and you don't want to have to redefine them all.

```
variants: {
    color: (color) => {
      return `text-${color}`
    }
}
```

**Comment by joe-bell** [comment]
It’s something I’ve been thinking about for sure, but Tailwind doesn’t support dynamic strings with its JIT engine (so it wouldn’t work for your use case unless you manually declare those strings elsewhere)

I’m cautious to add it for that reason
